engine.go

```diff
@@ -160,6 +165,61 @@ func (e *Engine) RunScript(scriptFile string, data any) error {
 	return nil
 }
 
+// MethodExecutor returns an execution function that enables calls to global template methods from easytemplate.
+func (e *Engine) MethodExecutor(scriptFile string, data any) (func(fnName string, args ...interface{}) (interface{}, error), error) {
```

Comment:
I was actually just thinking of having a `Run` that takes a method name and then someone would call `RunScript` first followed by `Run` to run additional methods in the environment setup by `RunScript`.

That would maybe simplify the implementation here

Reply:
We could do that but don't we need to do this in RunScript which we don't right now

result := esbuild.Transform(string(script), esbuild.TransformOptions{
Target:    esbuild.ES2015,
Loader:    esbuild.LoaderTS,
Sourcemap: esbuild.SourceMapExternal,
})

Reply:
https://github.com/speakeasy-api/easytemplate/blob/main/engine.go#L156

I'm not sure what the implications of adding that for every RunScript would be.